tests/queries/0_stateless/02346_text_index_preprocessor.sql

```diff
@@ -0,0 +1,73 @@
+SET allow_experimental_full_text_index = 1;
+SET use_skip_indexes_on_data_read = 1;
+
+-- Basic test preprocessor clause usage with lower
+DROP TABLE IF EXISTS tab;
+
+SELECT 'Test single tokenizer and preprocessor argument.';
+
+CREATE TABLE tab
+(
+    key UInt64,
+    str String,
+    INDEX idx(str) TYPE text(tokenizer = 'splitByNonAlpha', preprocessor = lower(str))
+)
+ENGINE = MergeTree
+ORDER BY tuple();
+
+INSERT INTO tab VALUES (1, 'foo'), (2, 'BAR'), (3, 'Baz');
+
+SELECT count() FROM tab WHERE hasToken(str, 'foo');
+SELECT count() FROM tab WHERE hasToken(str, 'FOO');
+
+SELECT count() FROM tab WHERE hasToken(str, 'BAR');
```

Comment:
L. 23 - 27 test the same thing as l. 20-21.

Reply:
Not exactly, see the new tests with a different preprocessor to find the differences